app/Http/Controllers/Api/V2/TeamPublicationController.php

```diff
@@ -0,0 +1,907 @@
+     *             @OA\Property(property="publication_type", type="string", example="Journal article, Book"),
+     *             @OA\Property(property="journal_name", type="string", example="A Journal"),
+     *             @OA\Property(property="abstract", type="string", example="A long description of the paper"),
+     *             @OA\Property(property="url", type="string", example="http://example"),
+     *             @OA\Property(property="mongo_id", type="string", example="38873389090594430"),
+     *             @OA\Property(property="status", type="string", enum={"ACTIVE", "DRAFT", "ARCHIVED"}),
+     *             @OA\Property(property="datasets", type="array",
+     *                @OA\Items(type="object",
+     *                   @OA\Property(property="id", type="integer"),
+     *                   @OA\Property(property="link_type", type="string"),
+     *                   @OA\Property(property="description", type="string"),
+     *                )
+     *             ),
+     *             @OA\Property(property="tools", type="array", example="[]", @OA\Items()),
+     *          ),
+     *       ),
+     *    ),
+     *    @OA\Response(
+     *       response="200",
+     *       description="Success response",
+     *       @OA\JsonContent(
+     *          @OA\Property(
+     *             property="message", type="string", example="success"),
+     *          @OA\Property(
+     *             property="data",
+     *             type="array",
+     *             example="[]",
+     *             @OA\Items(
+     *                type="array",
+     *                @OA\Items()
+     *             )
+     *          ),
+     *       ),
+     *    ),
+     *    @OA\Response(
+     *        response=400,
+     *        description="Error",
+     *        @OA\JsonContent(
+     *            @OA\Property(property="message", type="string", example="bad request"),
+     *        )
+     *    ),
+     *    @OA\Response(
+     *        response=401,
+     *        description="Unauthorized",
+     *        @OA\JsonContent(
+     *            @OA\Property(property="message", type="string", example="unauthorized")
+     *        )
+     *    ),
+     *    @OA\Response(
+     *        response=500,
+     *        description="Error",
+     *        @OA\JsonContent(
+     *            @OA\Property(property="message", type="string", example="error"),
+     *        )
+     *    )
+     * )
+     *
+     * @param  UpdatePublicationByTeamIdById  $request
+     * @param  int  $teamId
+     * @param  int  $id
+     * @return JsonResponse
+     */
+    public function update(UpdatePublicationByTeamIdById $request, int $teamId, int $id): JsonResponse
+    {
+        $input = $request->all();
+        $jwtUser = array_key_exists('jwt_user', $input) ? $input['jwt_user'] : [];
+        $initPublication = Publication::withTrashed()->where('id', $id)->first();
+        $this->checkAccess($input, null, $initPublication->owner_id, 'user');
```

Comment:
Should this be a team access check? e.g.
```suggestion
        $this->checkAccess($input, $initPublication->team_id, null, 'team'););
```

Reply:
updated